src/cljs/rems/administration/change_catalogue_item_form.cljs

```diff
@@ -0,0 +1,140 @@
+            [rems.atoms :as atoms :refer [readonly-checkbox document-title]]
+            [rems.collapsible :as collapsible]
+            [rems.dropdown :as dropdown]
+            [rems.flash-message :as flash-message]
+            [rems.table :as table]
+            [rems.text :refer [localize-time text get-localized-title]]
+            [rems.util :refer [navigate! fetch post!]]))
+
+(rf/reg-event-fx
+ ::enter-page
+ (fn [{:keys [db]} [_ catalogue-items]]
+   {:db (-> (dissoc db ::form)
+            (assoc ::catalogue-items catalogue-items))
+    ::fetch-forms nil}))
+
+(rf/reg-sub ::catalogue-items (fn [db _] (::catalogue-items db)))
+(rf/reg-sub ::form (fn [db _] (::form db)))
+(rf/reg-event-db ::set-form (fn [db [_ form]] (assoc-in db [::form] form)))
+
+(rf/reg-event-db
+ ::update-catalogue-item
+ (fn [db [_ old-catalogue-item-id new-catalogue-item-id new-form]]
+   (update db ::catalogue-items (fn [items]
+                                  (for [item items]
+                                    (if (= (:id item) old-catalogue-item-id)
+                                      (assoc item :id new-catalogue-item-id :formid (:form/id new-form) :form-name (:form/title new-form))
+                                      item))))))
+
+
+(rf/reg-event-fx
+ ::change-catalogue-item-form!
+ (fn [{:keys [db]} [_ catalogue-item-id form on-success]]
+   (post! (str  "/api/catalogue-items/" catalogue-item-id "/change-form")
+          {:params {:form (:form/id form)}
+           :handler (fn [result]
+                      (rf/dispatch [::update-catalogue-item catalogue-item-id (:catalogue-item-id result) form])
+                      (rf/dispatch [:rems.table/toggle-row-selection {:id :rems.administration.catalogue-items/catalogue} catalogue-item-id])
+                      (rf/dispatch [:rems.table/toggle-row-selection {:id :rems.administration.catalogue-items/catalogue} (:catalogue-item-id result)])
+                      (on-success))
+           :error-handler (flash-message/default-error-handler :top [text :t.administration/change-form])})
+   {}))
+
+(defn- fetch-forms []
+  (fetch "/api/forms"
+         {:handler #(rf/dispatch [::fetch-forms-result %])
+          :error-handler (flash-message/default-error-handler :top "Fetch forms")}))
+
+(rf/reg-fx ::fetch-forms fetch-forms)
+
+(rf/reg-event-db ::fetch-forms-result (fn [db [_ forms]] (assoc db ::forms forms)))
+
+(rf/reg-sub ::forms (fn [db _] (::forms db)))
+
+(defn- form-change-loop [items form]
+  (let [item (first items)]
+    (cond (empty? items)
+          (flash-message/show-default-success! :top [text :t.administration/change-form])
+
+          (not= (:formid item) (:form/id form))
+          (rf/dispatch [::change-catalogue-item-form! (:id item) form (partial form-change-loop (rest items) form)])
+
+          :else (recur (rest items) form))))
+
+(defn- all-items-have-the-form-already? [items form]
+  (every? (comp #{(:form/id form)} :formid) items))
```

Comment:
I'm having a hard time understanding this line of code, since it uses a set as a function together with comp, both of which are complicated enough to understand alone.

Reply:
If I understand correctly, this could be refactored to `#(= (:form/id form) (:formid %))`

Reply:
Using a set for this is idiomatic usage in Clojure.  I think using `comp` is also idiomatic here. However the alternative is also pretty clear so I'll change it.